src/Rest/Helpers/Channel/Message/AddAttachment.php

```diff
@@ -8,17 +8,31 @@
 
 trait AddAttachment
 {
+    /** @var AttachmentBuilder[] */
+    private array $attachments;
```

Comment:
```suggestion
    private array $attachments = [];
```
Why not have this defined by default? Could just use `empty` in your `getAttachments` and `hasAttachments` methods instead `isset` repeatedly in `addAttachment`.

Reply:
Not sure if it applies here, just did it to be save.

Let's say you're editing a message, if the original message has attachments, you may want to edit the content only. You add `content` to the payload with the updated text. If you then have an empty array `attachments` in the payload it could remove the attachments there are in the original message.

You may also want to remove the attachments in such a scenario, hence the need for 2 states. 
- No attachments at all
- Empty attachments